test/truefi2/lines-of-credit/TrueRateAdjuster.test.ts

```diff
@@ -414,6 +423,49 @@ describe('TrueRateAdjuster', () => {
     )
   })
 
+  describe('totalTVL', () => {
+    const YEAR = DAY * 365
+    let loan
+    let lender
+
+    beforeEach(async () => {
+      const { rater, tru, stkTru, loanFactory, standardPool: pool, lender: _lender, standardToken: tusd } = await setupTruefi2(owner, provider)
+      loan = await createApprovedLoan(rater, tru, stkTru, loanFactory, borrower, pool, 1_000_000, YEAR, 1000, owner, provider)
+      lender = _lender
+
+      const mockPool1 = await deployMockContract(owner, ITrueFiPool2WithDecimalsJson.abi)
+      const mockPool2 = await deployMockContract(owner, ITrueFiPool2WithDecimalsJson.abi)
+      await rateAdjuster.addPoolToTVL(pool.address)
+      await rateAdjuster.addPoolToTVL(mockPool1.address)
+      await rateAdjuster.addPoolToTVL(mockPool2.address)
+
+      await tusd.mint(owner.address, parseEth(1e7))
+      await tusd.approve(pool.address, parseEth(1e7))
+      await pool.join(parseEth(1e7))
+      await mockPool1.mock.decimals.returns(18)
+      await mockPool1.mock.poolValue.returns(parseEth(1e7))
+      await mockPool2.mock.decimals.returns(18)
+      await mockPool2.mock.poolValue.returns(parseEth(1e7))
+    })
+
+    it('totalTVL returns sum of poolValues of all pools with 18 decimals precision', async () => {
+      expect(await rateAdjuster.totalTVL(18)).to.equal(parseEth(3e7))
+    })
+
+    it('totalTVL remains unchanged after borrowing', async () => {
+      expect(await rateAdjuster.totalTVL(18)).to.equal(parseEth(3e7))
+      await lender.connect(borrower).fund(loan.address)
+      expect(await rateAdjuster.totalTVL(18)).to.equal(parseEth(3e7))
+    })
+
+    it('totalTVL scales with loan interest', async () => {
+      expect(await rateAdjuster.totalTVL(18)).to.equal(parseEth(3e7))
+      await lender.connect(borrower).fund(loan.address)
+      await timeTravel(provider, YEAR)
+      expectScaledCloseTo(await rateAdjuster.totalTVL(18), parseEth(3e7).add(parseEth(1)))
+    })
+  })
```

Comment:
Maybe let's also add the test that checks how adding a new pool affects the TVL?